src/rust/ensogl/lib/core/src/animation/frp/animation.rs

```diff
@@ -0,0 +1,158 @@
+        // FIXME[WD]: The precision should become default and should be increased in all simulators
+        //            that work with pixels. The reason is that by default the simulator should
+        //            give nice results for animations in the range of 0 .. 1, while it should not
+        //            make too many steps when animating bigger values (like pixels).
+        simulator.set_precision(0.001);
+        frp::extend! { network
+            target    <- any_mut::<T>();
+            precision <- any_mut::<f32>();
+            skip      <- any_mut::<()>();
+            eval  target    ((t) simulator.set_target_value(mix::into_space(t.clone())));
+            eval  precision ((t) simulator.set_precision(*t));
+            eval_ skip      (simulator.skip());
+        }
+        let value = value_src.into();
+        network.store(&simulator);
+        Self{target,precision,value,skip}
+    }
+
+    /// Constructor. The initial value is provided explicitly.
+    pub fn new_with_init(network:&frp::Network, init:T) -> Self {
+        let this = Self::new(network);
+        this.target.emit(init);
+        this.skip.emit(());
+        this
+    }
+
+    /// Constructor. There is no initial value. The first emitted `target` value will be used
+    /// without animation.
+    pub fn new_non_init(network:&frp::Network) -> Self {
+        let this = Self::new(network);
+        frp::extend! { network
+            init      <- any_mut();
+            on_init   <- this.target.gate_not(&init);
+            init      <+ this.target.constant(true);
+            this.skip <+ on_init.constant(());
+        }
+        this
+    }
+}
+
+
+
+// ============================
+// === DEPRECATED Animation ===
+// ============================
+
+/// Smart animation handler. Contains of dynamic simulation and frp endpoint. Whenever a new value
+/// is computed, it is emitted via the endpoint.
+///
+/// # DEPRECATION
+/// This component is deprecated. Use `Animation` instead, which exposes much more FRP-oriented API
+/// than this component. The transition to new version should be straightforward but requires some
+/// attention. The functionalities should be the same and should be accessible by similar API with
+/// two differences:
+///
+/// 1. The API bases on FRP now, which means that the usage can probably be refactored to look
+///    much nicer.
+///
+/// 2. After setting the value for the first time, the value is provided as the output value without
+///    any animation. This is different behavior from the previous implementation, where even
+///    setting the value for the first time would create animation between `default()` value and the
+///    new target. If your code depends on this behavior, it needs to be changed.
+#[derive(CloneRef,Derivative,Debug,Shrinkwrap)]
+#[derivative(Clone(bound=""))]
+#[allow(missing_docs)]
+#[allow(non_camel_case_types)]
+pub struct DEPRECATED_Animation<T:mix::Mixable> {
```

Comment:
Should we use the [actual deprecated attribute](https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html) here? In any case, it would be good to have a maintenance issue about removing these.

Reply:
We were chatting about it with @farmaazon and the depreciation attribute would not give us any benefits - we would disable the warnings because we want the CI to pass, and such a name is actually ore visible on call side.